### PR TITLE
chore: prettify types

### PR DESCRIPTION
**Pull request by Raiden1411**
## Additional Information

- [x] I read the [contributing guide](https://github.com/wagmi-dev/abitype/blob/main/.github/CONTRIBUTING.md)
- [x] I added documentation related to the changes made.
- [x] I added or updated tests related to the changes made.

Your ENS/address:

---

## PR-Codex overview
This PR adds the `AbiEventParameter` type and makes various changes to improve the `abitype` library.

### Detailed summary
- Adds the `AbiEventParameter` type
- Improves the `AbiParameter` type
- Adds `AbiError` to `src/index.ts`
- Adds `AbiEventParameter` to `src/index.ts` and `src/zod/index.ts`
- Adds `AbiEventParameter` and `AbiError` to `src/zod/index.test.ts`
- Updates `vitest.config.ts` to exclude `src/**/index.ts`
- Updates `src/errors.ts` to use `Prettify` and `OneOf` types
- Adds `AbiReceive` to `src/abi.ts` and `src/zod/zod.ts`
- Adds `AbiEvent` and `AbiError` to `src/zod/zod.ts`
- Adds various Solidity types to `src/zod/zod.ts`
- Adds tests for various types to `src/zod/zod.test.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: .changeset/slow-olives-occur.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"abitype": patch
+---
+
+Prettified types.
+Created new `AbiEventParameters` type.
```
Comment (tmm) [inline .changeset/slow-olives-occur.md:6]:
```suggestion
Created new `AbiEventParameter` type.
```

**Review by tmm — COMMENTED** [review]
Great idea! Want to add the matching Zod schema for `AbiEventParameter` too?

**Comment by Raiden1411** [comment]
Will do!

File: src/zod/index.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ export {
   Abi,
   AbiConstructor,
   AbiEvent,
+  AbiEventParameter,
```
Comment (tmm) [inline src/zod/index.ts:5]: Let's add to the docs too: https://abitype.dev/api/zod.html#schemas

Comment (Raiden1411) [reply src/zod/index.ts:5]: Sorry my bad. Just added it.

**Comment by tmm** [comment]
Waiting to push change that boosts coverage, but GitHub is degraded right now.

**Pull request merged**